### PR TITLE
speed up ci by splitting off eval and build + fix ci with restricted namespaces

### PR DESCRIPTION
**Pull request by Mic92**
- This speeds up macOS builds from 30 minutes to 11 minutes (3x faster).        
 - Also improve error reporting e.g. printing out what actually failed to build. 
 - As a result we also no longer need swap.
 - The resulting script is also very useful for running checks locally as it does not need so much RAM as before (we used significantly more memory than what our nixpkgs eval ci these days consume)

### CODE REVIEW

**Comment by Mic92** [comment]
Now I will have a look at the CI issue.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -33,25 +41,10 @@ jobs:
...
-        (
-          while sleep 60; do
-            free -h
-          done
-        ) &
-    - run: nix --experimental-features 'nix-command flakes' flake check -L
-    - run: nix --experimental-features 'nix-command flakes' flake show --all-systems --json
+    # Since ubuntu 22.30, unprivileged usernamespaces are no longer allowed to map to the root user:
+    # https://ubuntu.com/blog/ubuntu-23-10-restricted-unprivileged-user-namespaces
+    - run: sudo sysctl -w kernel.apparmor_restrict_unprivileged_userns=0
```
Comment (Mic92) [inline .github/workflows/ci.yml:46]: I want to investigate if we actually need `--map-root-user` in our tests.

Comment (Mic92) [reply .github/workflows/ci.yml]: There is no way around without switching to a different sandbox implementation i.e. bubblewrap.

**Review by tomberek — APPROVED** [review]

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**